dotcom-rendering/src/web/layouts/NewsletterSignupLayout.tsx

```diff
@@ -101,6 +115,77 @@ const leftColWrapperStyle = css`
 	}
 `;
 
+const previewButtonWrapperStyle = css`
+	padding: ${space[2]}px 0;
+	${from.desktop} {
+		display: none;
+	}
+`;
+
+const mainGraphicWrapperStyle = css`
+	border-radius: ${space[2]}px;
+	overflow: hidden;
+	margin: ${space[4]}px 0;
+
+	& > figcaption {
+		display: flex;
+
+		${until.desktop} {
+			display: none;
+		}
+
+		background-color: ${brandAlt[400]};
+		align-items: center;
+		padding: ${space[1]}px;
+		${textSans.medium({ fontWeight: 'bold', lineHeight: 'tight' })}
+
+		svg {
+			margin-right: ${space[1]}px;
+			flex-shrink: 0;
+		}
+	}
+`;
+
+interface NewsletterDetailProps {
+	text: string;
+	showUntil?: Breakpoint;
+}
+const NewsletterDetail = ({ text, showUntil }: NewsletterDetailProps) => {
+	const displayRule = showUntil
+		? `
+	display: none;
+	${until[showUntil]} {
+		display: flex;
+	}
+	`
+		: `display: flex;`;
+
+	const containerStyle = css`
+		${displayRule}
+		align-items: center;
+		padding-top: ${space[3]}px;
+		padding-bottom: ${space[2]}px;
+
+		svg {
+			background-color: ${brandAlt[400]};
+			border-radius: 50%;
+			padding: 2px;
+			margin-right: ${space[2]}px;
+		}
+
+		span {
+			${textSans.medium({ fontWeight: 'bold' })};
+		}
+	`;
+
+	return (
+		<div css={containerStyle}>
+			<SvgEnvelope size="small" />
+			<span>{text}</span>
+		</div>
+	);
+};
+
 interface Props {
```

Comment:
Nitpick: There's a convention to keep the props for a component at the top of the file. It can help when opening new files and trying to understand them

Reply:
Before. Most components have the first line after the imports as `type Props = ...`